workflow/usage.rst

```diff
@@ -450,3 +450,216 @@ The following example shows these functions in action:
+                        <framework:from>draft</framework:from>
+                        <framework:to>review</framework:to>
+                        <framework:metadata>
+                            <framework:priority>0.5</framework:priority>
+                        </framework:metadata>
+                    </framework:transition>
+                    <!-- ... -->
+                </framework:workflow>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/workflow.php
+
+        $container->loadFromExtension('framework', array(
+            // ...
+            'workflows' => array(
+                'blog_publishing' => array(
+                    'metadata' => array(
+                        'title' => 'Blog Publishing Workflow',
+                    ),
+                    // ...
+                    'places' => array(
+                        'draft' => array(
+                            'metadata' => array(
+                                'max_num_of_words' => 500,
+                            ),
+                        ),
+                        // ...
+                    ),
+                    'transitions' => array(
+                        'to_review' => array(
+                            'from' => 'draft',
+                            'to' => 'review',
+                            'metadata' => array(
+                                'priority' => 0.5,
+                            ),
+                         ),
+                     ),
+                 ),
+             ),
+         ));
+
+Then, you can access this metadata in your PHP code as follows:
+
+In your Controller::
+
+    public function myControllerAction(Registry $registry, Article $article)
+    {
+        $workflow = $registry->get($article);
+
+        // Or, if you don't inject the Workflow Registry, fetch from the Container:
+        $workflow = $this->get('workflow.article');
+
+        $workflow
+            ->getMetadataStore()
+            ->getWorkflowMetadata()['title'] ?? false
+        ;
+
+        // or
+        $workflow
+            ->getMetadataStore()
+            ->getPlaceMetadata('draft')['title'] ?? false
+        ;
+
+        // or
+        $aTransition = $workflow->getDefinition()->getTransitions()[0];
+        $workflow
+            ->getMetadataStore()
+            ->getTransitionMetadata($aTransition)['title'] ?? false
+        ;
+    }
+
+There is a shortcut that works with everything::
+
+    $workflow
+        ->getMetadataStore()
+        ->getMetadata('title')
+    ;
+
+In a Flash message in your Controller::
+
+            // $transition = ...; (an instance of Transition)
+            $title = $this->get('workflow.article')->getMetadataStore()->getMetadata('title', $transition);
+            $request->getSession()->getFlashBag()->add('info', "You have successfully applied the transition with title: '$title'");
+
+In a listener, access via the Event::
```

Comment:
Maybe we could add a little introduction for this very nice example.

Some context about what it is doing and why it's useful to access metadata in there, also that we use a variant here `Event::getMetadata()`. What do you think?

Reply:
Those are all very good ideas 👍

Reply:
Added in https://github.com/symfony/symfony-docs/pull/11209/commits/0affc6b8a37ccf7ffcb141d960f3d20aa187766c.